docs/ARCHITECTURE.md

```diff
@@ -254,10 +362,14 @@ direction LR
             +generateSpeechOperation() GenerativeAiOperation
             +generateEmbeddingsOperation() EmbeddingOperation
             +generateText() string
+            +generateTexts(int $candidateCount) string[]
             +streamGenerateText() Generator< string >
             +generateImage() File
+            +generateImages(int $candidateCount) File[]
             +convertTextToSpeech() File
+            +convertTextToSpeeches(int $candidateCount) File[]
             +generateSpeech() File
+            +generateSpeeches(int $candidateCount) File[]
```

Comment:
Should we clarify the `int $candidateCount` as optional, i.e. `?int $candidateCount`? If someone already went with `usingCandidateCount($candidateCount)` in the builder before, there wouldn't be a need to specify it here again. In other words, it's only a short-cut, as discussed before.

On the other hand, if you didn't already use `usingCandidateCount`, it would be weird not to specify it here. Although, that's kind of similar if you did `generateTextResult()->candidates()` without having specified a candidate count. I think if you don't specify it, we'd just stick with the default of 1.

WDYT?